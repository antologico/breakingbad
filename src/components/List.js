import React, { useState } from 'react'
import PropTypes from 'prop-types'
import { Title, Input, Characteres } from '../ui'
import { injectIntl } from 'react-intl'
import imdb from '../images/imdb.svg'
import CharacterItem from './CharacterItem'

const List = ({ characteres, intl }) => {
  const [filter, setFilter] = useState('')
  const filtered = filter !== ''
    ? characteres.filter(({ data }) => data.name.includes(filter))
    : characteres

  return <>
    <Title><img src={imdb} alt='IMBd' height='30px' /> Breaking Bad</Title>
    <div>
        <Input
          data-test='search-box'
          placeholder={'ðŸ”Ž ' + intl.formatMessage({  id: 'example', defaultMessage: 'Ex, Walter' },)}
          width='100%'
          value={filter}
          onChange={(e) => setFilter(e.target.value)} />
    </div>
    <Characteres>
      {filtered && filtered.map(({ data }, index) => <CharacterItem key={`item_${index}`} { ...data } />)}
    </Characteres>
  </>
}

List.propTypes = {
  intl: PropTypes.object,
  characteres: PropTypes.arrayOf(PropTypes.shape({
    name: PropTypes.string,
    image: PropTypes.string,
  }))
}

export default injectIntl(List)
