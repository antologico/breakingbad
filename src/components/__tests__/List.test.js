import React from 'react'
import { create, mount } from '../../tests/utils'
import List from '../List'

jest.mock('react-router-dom', () => ({
  Link: ({children}) => <a id='link'>{children}</a>
}))

jest.mock('../../images/imdb.svg', () => 'image')

const characteres = [
  { data: { id: 1, name: 'Alt', image: 'image1.png' } },
  { data: { id: 2, name: 'Control', image: 'image1.png' } },
]

describe(' component', () => {
  test('renders properly', () => {
    const component = create(
        <List characteres={characteres} />
    )
    expect(component.toJSON()).toMatchSnapshot()
  })
  test('filter properly', () => {
    const component = mount(
        <List characteres={characteres} />
    )
    component.update()
    expect(component.find('[data-test="character_1"]').exists()).toBe(true)
    expect(component.find('[data-test="character_2"]').exists()).toBe(true)
    
    const input = component.find('input').at(0)
    expect(input.exists()).toBe(true)
    input.simulate('change', { target: { value: 'Alt'} })
   
    component.update()
    expect(component.find('[data-test="character_1"]').exists()).toBe(true)
    expect(component.find('[data-test="character_2"]').exists()).toBe(false)
  })
})