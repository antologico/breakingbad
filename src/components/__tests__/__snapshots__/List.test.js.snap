// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` component renders properly 1`] = `
Array [
  <h1
    className="sc-bczRLJ ibOPpr"
  >
    <img
      alt="IMBd"
      height="30px"
      src="image"
    />
     Breaking Bad
  </h1>,
  <div>
    <input
      className="sc-kDDrLX bHKnbf"
      data-test="search-box"
      onChange={[Function]}
      placeholder="ðŸ”Ž Buscar: Walter White"
      value=""
      width="100%"
    />
  </div>,
  <div
    className="sc-iqcoie gtVNTc"
  >
    <nav
      data-test="character_1"
      data-test-group="character-item"
    >
      <a
        id="link"
      >
        <span>
          Alt
        </span>
        <img
          alt="Alt"
          className="sc-eCYdqJ dJkwBv"
          height="40px"
          src="image1.png"
          width="40px"
        />
      </a>
    </nav>
    <nav
      data-test="character_2"
      data-test-group="character-item"
    >
      <a
        id="link"
      >
        <span>
          Control
        </span>
        <img
          alt="Control"
          className="sc-eCYdqJ dJkwBv"
          height="40px"
          src="image1.png"
          width="40px"
        />
      </a>
    </nav>
  </div>,
]
`;
